### PR TITLE
refactor(iterators): correctly type iterators.py

### LINKED ISSUE #836: Several classes in iterators.py are untyped
### Summary

There are several classes in iterators.py where their methods and attributes have no type hints defined.

### What is the feature request for?

The core library

### The Problem

As stated in #816, not typing methods and attributes in classes makes it hard to tell for static type checkers to infer their types.

### The Ideal Solution

Type all of the methods of the following classes:

- `_ChunkedAsyncIterator`
- `_MappedAsyncIterator`
- `_FilteredAsyncIterator`
- `ReactionIterator`
- `HistoryIterator`
- `AuditLogIterator`
- `GuildIterator`
- `MemberIterator`
- `create_event` of `ScheduledEventIterator`
- `create_user` of `ScheduledEventUserIterator`

It also looks like a lot of the attributes in all of the classes defined in iterators.py have no type hints.

### The Current Solution

_No response_

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by teaishealthy**
## Summary

Closes #836 

This PR also adds the `pyright: strict` comment to iterators.py causing pyright to check the file with the strict ruleset.

Adding this will ensure that iterators typehint qualitiy does not regress.

## Checklist

- [ ] If code changes were made then they have been tested.
  - [ ] I have updated the documentation to reflect the changes.
  - [x] I have run `task pyright` and fixed the relevant issues.
- [x] This PR fixes an issue.
- [ ] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

File: nextcord/iterators.py [hunk 1]
```diff
@@ -210,16 +220,16 @@ async def next(self) -> Union[User, Member]:
...
         from .user import User
 
         if self.limit > 0:
             retrieve = self.limit if self.limit <= 100 else 100
 
             after = self.after.id if self.after else None
-            data: List[PartialUserPayload] = await self.getter(
-                self.channel_id, self.message.id, self.emoji, retrieve, after=after
+            data: List[PartialUserPayload] = await self.state.http.get_reaction_users(
+                self.channel_id, self.message.id, self.emoji, retrieve, after=after  # type: ignore
```
Comment (EmreTech) [inline nextcord/iterators.py:232]: Also leave a comment here

**Review by EmreTech — APPROVED** [review]
Everything else looks good

File: nextcord/iterators.py [hunk 2]
```diff
@@ -117,7 +127,7 @@ def chunk(self, max_size: int) -> _ChunkedAsyncIterator[T]:
         return _ChunkedAsyncIterator(self, max_size)
 
     def map(self, func: _Func[T, OT]) -> _MappedAsyncIterator[OT]:
-        return _MappedAsyncIterator(self, func)
+        return _MappedAsyncIterator(self, func)  # type: ignore
```
[the remark this answers is not in the document]
Comment (teaishealthy) [reply nextcord/iterators.py:130]: Was not typed accurately, fixed now.

**Review by EmreTech — COMMENTED** [review]
One last thing

**Review by EmreTech — APPROVED** [review]

**Comment by teaishealthy** [comment]
Failing pyright would be fixed by a pyright bump

**Comment by DenverCoder1** [comment]
> Failing pyright checks would be fixed by a pyright version bump

You should be able to bump the version in `/requirements-dev.txt`, right?

**Pull request closed without merging**